vou/utils.py

```diff
@@ -22,3 +23,13 @@ def load_json(json_file: Path):
     """
     with open(json_file) as f:
         return json.load(f)
+
+def weighted_random_by_dct(dct):
+    rand_val = random.random()
```

Comment:
In response to the version from https://github.com/RTIInternational/virtual-opioid-user/pull/30/commits/d9c9953be02637024fa0414cfb234b8d78f7974c:

I suggest passing the random number generator itself as an argument rather than a random value. This would help prevent future errors/misuse, since there's nothing in the current structure enforcing that the value passed is a random number from `simulation.rng`. 

Here this would look like
```
def weighted_random_by_dct(dct: dict, rng: random.Random):
    rand_val = rng.random()
    ...
```

And the function call would look like
```
weighted_random_by_dct(self.person.drug_params['admin_mode_distributions'][self.dose_type], self.rng)
```